src/ucp/core/ucp_ep.c

```diff
@@ -1201,6 +1218,13 @@ static void ucp_ep_discard_lanes(ucp_ep_h ep, ucs_status_t discard_status)
         }
 
         ucs_debug("ep %p: discard uct_ep[%d]=%p", ep, lane, uct_ep);
+
+        if (!ucp_wireup_ep_test(uct_ep) ||
+            ((ucp_wireup_ep(uct_ep)->super.uct_ep != NULL) &&
+            ucp_wireup_ep(uct_ep)->super.is_owner)) {
```

Comment:
Use a Macro or function for the conditions.

Reply:
done